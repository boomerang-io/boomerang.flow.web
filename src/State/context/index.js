import React from "react";

export const AppContext = React.createContext();

export const ExecutionContext = React.createContext({
  tasks: [],
  workflowExecution: {},
  workflowRevision: {},
});

export const EditorContext = React.createContext({
  revisionDispatch: () => {},
  revisionState: {},
  revisionQuery: {},
  summaryState: {},
  taskTemplatesData: [],
});
