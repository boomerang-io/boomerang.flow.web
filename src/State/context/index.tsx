import React from "react";
import { FlowTeam, FlowUser } from "Types";

export function createContext<ContextType>() {
  const context = React.createContext<ContextType | undefined>(undefined);
  function useContext() {
    const contextValue = React.useContext(context);
    if (!contextValue) throw new Error("useContext must be inside a Provider with a value");
    return contextValue;
  }
  return [useContext, context.Provider] as const;
}

export const [useAppContext, AppContextProvider] = createContext<AppContext>();

type AppContext = {
  isTutorialActive: boolean;
  setIsTutorialActive: (isActive: boolean) => void;
  user: FlowUser;
  teams: FlowTeam[];
};

interface taskProvider {
  category: string;
  id: string;
  icon: any;
  name: string;
}

interface stepInterface {
  activityId: string;
  duration: number;
  flowTaskStatus: string;
  id: string;
  order: number;
  outputs: any;
  startTime: string;
  taskId: string;
  taskName: string;
}

interface workflowExecutionInterface {
  creationDate: string;
  duration: number;
  id: string;
  properties: any; //array, may need to be changed later
  status: string;
  steps: Array<stepInterface>;
  teamName: string;
  trigger: string;
  workflowId: string;
  workflowRevisionid: string;
}

// export const ExecutionContext = React.createContext({});
interface executionContextInterface {
  tasks?: Array<taskProvider>;
  workflowExecution?: workflowExecutionInterface;
  workflowRevision?: object;
}
// export const ExecutionContext = React.createContext<executionContextInterface>({});
export const ExecutionContext = React.createContext<Partial<executionContextInterface>>({});

// export const EditorContext = React.createContext({});

interface revisionInterface {
  changelog: object;
  config: object;
  dag: {
    gridSize: number;
    id: string;
    links: Array<{
      color: string;
      curvyness: number;
      executionCondition: string;
      extras: object;
      id: string;
      labels: Array<string>; //i think this type is right?
      linkId: string;
      selected: false;
      source: string;
      sourcePort: string;
      switchCondition: string | null;
      target: string;
      targetPort: string;
      type: string;
      width: number;
    }>;
    nodes: Array<{
      extras: {};
      id: string;
      nodeId: string;
      passedName: string;
      ports: Array<{
        id: string;
        links: Array<string>;
        name: string;
        nodePortId: string;
        position: string;
        selected: boolean;
        type: string;
      }>;
      selected: boolean;
      templateUpgradeAvailable: boolean;
      type: string;
      x: number;
      y: number;
    }>;
    offsetX: number;
    offsetY: number;
    zoom: number;
  };
  id: string;
  templateUpgradesAvailable: string;
  version: number;
  workFlowId: string;
}

interface summaryInterface {
  data: {
    description: string;
    enableACCIntegration: boolean;
    enablePersistentStorage: boolean;
    flowTeamId: string;
    icon: string;
    id: string;
    name: string;
    properties: Array<object>;
    revisionCount: number;
    shortDescription: string;
    status: string;
    templateUpgradesAvailable: boolean;
    triggers: object;
  };
}

interface taskInterface {
  category: string;
  createdDate: string;
  currentVersion: number;
  description: string;
  icon: string;
  id: string;
  lastModified: string;
  name: string;
  nodeType: string;
  revisions: Array<object>;
  status: string;
}

interface editorContextInterface {
  revisionDispatch?: Function;
  revisionState: revisionInterface;
  summaryQuery: summaryInterface;
  taskTemplatesData: Array<taskInterface>;
}

export const EditorContext = React.createContext<Partial<editorContextInterface>>({});
