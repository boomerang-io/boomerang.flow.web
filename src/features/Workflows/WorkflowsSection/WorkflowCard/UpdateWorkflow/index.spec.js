import React from "react";
import { actions as importWorkflowActions, initialState } from "State/importWorkflow";
import ImportWorkflow from ".";
import { fireEvent } from "@testing-library/react";

const mockfn = jest.fn();

const props = {
  fetchTeams: mockfn,
  handleImportWorkflow: mockfn,
  importWorkflowActions
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("ImportWorkflow --- Snapshot Test", () => {
  it("Capturing Snapshot of ImportWorkflow", () => {
    const { baseElement, getByText } = rtlReduxRender(<ImportWorkflow {...props} />, {
      initialState: { importWorkflow: initialState }
    });
    fireEvent.click(getByText(/Choose a file or drag one here/i));
    expect(baseElement).toMatchSnapshot();
  });
});
