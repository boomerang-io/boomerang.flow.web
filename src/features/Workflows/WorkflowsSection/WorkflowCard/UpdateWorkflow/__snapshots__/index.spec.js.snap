// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportWorkflow --- Snapshot Test Capturing Snapshot of ImportWorkflow 1`] = `
<body
  class="bx--bmrg-body-modal-is-open"
  id="app"
>
  <div />
  <div
    class="ReactModalPortal"
  >
    <div
      class="bx--bmrg-modal-overlay --is-open"
    >
      <div
        class="bx--bmrg-modal-container bx--bmrg-modal-flow-container container --is-open"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="bx--modal-header"
        >
          <p
            class="bx--modal-header__heading bx--type-beta"
          >
            Update .json file
          </p>
          <button
            class="bx--modal-close"
            title="Close the modal"
            type="button"
          >
            <svg
              aria-label="Close the modal"
              class="bx--modal-close__icon"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style="will-change: transform;"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
              />
            </svg>
          </button>
        </div>
        <form
          class="bx--bmrg-modal-flow-form"
        >
          <p
            class="bx--bmrg-modal-flow-form__title"
          >
            Update a Workflow - Select the Workflow file you want to upload
          </p>
          <div
            class="bx--modal-content"
            style="height: 21rem; width: 100%;"
          >
            <div
              class="bx--file"
            >
              <label
                class="bx--file-browse-btn"
                for="id2"
                tabindex="0"
              >
                <div
                  class="bx--file__drop-container"
                  role="button"
                >
                  Choose a file or drag one here
                  <input
                    accept=".json"
                    class="bx--file-input"
                    id="id2"
                    name="Workflow"
                    tabindex="-1"
                    type="file"
                  />
                </div>
              </label>
            </div>
            
          </div>
          <div
            class="bx--modal-footer"
          >
            <button
              class="bx--btn bx--btn--secondary"
              tabindex="0"
              type="button"
            >
              Cancel
            </button>
            <button
              class="bx--btn bx--btn--primary bx--btn--disabled"
              disabled=""
              tabindex="0"
              type="button"
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
